leader-election-impl/src/main/java/com/palantir/leader/proxy/AwaitingLeadershipProxy.java

```diff
@@ -219,7 +116,7 @@ protected Object handleInvocation(Object proxy, Method method, Object[] args) th
                     if (leading == StillLeadingStatus.NOT_LEADING || leading == StillLeadingStatus.NO_QUORUM) {
                         return Futures.submitAsync(
                                 () -> {
-                                    markAsNotLeading(leadershipToken, null /* cause */);
+                                    handleNotLeading(leadershipToken, null /* cause */);
```

Comment:
The difference here is that we are now not clearing the delegate. This is obviously fine for calls that have already gotten their maybeValidDelegate, and is fine for calls that have not gotten their leadership token. The difference is if we lose leadership after getting the token, but before getting the delegate. It is pretty obvious we had no guarantees there before though.

Reply:
In fact I'm pretty sure the preconditions check on the delegate below can now never fail